### ISSUE TITLE
[a11y]: Data-table selection aria-checked 

### ISSUE THREAD

**Issue by chubbywhale** [carbon-design-system/carbon#11532]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

@carbon/react 1.4.0

### React version

17.0.2

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker - Latest Deployment

### Description

The Accessibility Checker returns the following violation "The ARIA attribute 'aria-checked' is not valid for the element <input> with ARIA role 'checkbox'" with the select all checkbox in the headers when its `aria-checked="mixed"`

This was the HTML output:
`<input
    aria-checked="mixed"
    type="checkbox"
    name="select-all"
    id="data-table-1__select-all"
    class="cds--checkbox">`

<img width="343" alt="image" src="https://user-images.githubusercontent.com/37340587/171530499-e81f95ab-16d8-489e-9459-597f1c252c05.png">

<img width="646" alt="image" src="https://user-images.githubusercontent.com/37340587/171530523-fd3656fb-0d57-4c46-9bf5-bea2037190a8.png">

It probably needs to update the code similar to the Checkbox component where it no longer uses the `aria-checked` attribute.

### Reproduction/example

https://stackblitz.com/edit/github-uxyvty?file=src%2FApp.jsx

### Steps to reproduce

Just run the about sample and execute the Accessibility Check against it.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tomleah** [comment]
Hi @aledavila,

Is this something that would be able to be backported to Carbon 10?

**Comment by tw15egan** [comment]
@tomleah I can backport this quick for you, will go out in the next patch release of v10 when merged in 👍

**Comment by tomleah** [comment]
Amazing! Thank you @tw15egan 🚀

**Comment by Kenzku** [comment]
Hej @tw15egan , Could you please advise which release will this fix go into and estimated release date?

the carbon versions we have now:

```
    "@carbon/charts-react": "^0.58.2",
    "@carbon/ibm-products": "^1.22.0",
    "@carbon/icons-react": "^10.48.0",
    "@carbon/react": "^1.8.0",
    "@carbon/styles": "^1.8.0",
```

Thank you

**Comment by tw15egan** [comment]
It looks like it was in `@carbon/react` `v1.10.0` 

https://github.com/carbon-design-system/carbon/releases/tag/v11.9.1

**Comment by Kenzku** [comment]
@tw15egan 
I have two follow up:
1. do you suggest we only update `@carbon/react` ? or we shall update other carbons all together, as I do not see `ibm-products` and `charts-react` in the link
2. I only see `rc` version in the link you provided, e.g. `@carbon/react@1.10.0-rc.0`. Does the newer version: https://github.com/carbon-design-system/carbon/releases/tag/v11.15.0 compatible with our dependencies please?

**Comment by tw15egan** [comment]
I would try bumping them all to the latest and see if everything works as expected, as you will receive any bugfixes made between `1.8.0` and now

**Comment by Kenzku** [comment]
@tw15egan I chatted to the ibm-product team, detail at: https://github.com/carbon-design-system/ibm-cloud-cognitive/issues/2408#issuecomment-1292460792

In short, I was wondering do you have a fix for this AVT issue for Carbon V10 please?

**Comment by tw15egan** [comment]
Yes this was backported to v10 recently: https://github.com/carbon-design-system/carbon/pull/12106 and should be available in `7.59.4`

https://github.com/carbon-design-system/carbon/releases/tag/v10.59.4
